### PR TITLE
ipa_sudorule, ipa_hbacrule: change ipaenabledflag type to bool

### PR DESCRIPTION
**Pull request by aBUDmdBQ**
##### SUMMARY
freeipa changed the type of ipaenabledflag to bool with commit https://pagure.io/freeipa/c/6c5f2bcb301187f9844985ffe309c7d2262e16f3

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
- ipa_sudorule
- ipa_hbacrule

### CODE REVIEW

**Comment by felixfontein** [comment]
Hmm, this change will break with older FreeIPA versions. I guess a version check is needed?

**Comment by aBUDmdBQ** [comment]
Note: This is similar to changes in #7795. Should fix #5894 and #7082.

**Comment by aBUDmdBQ** [comment]
> Hmm, this change will break with older FreeIPA versions. I guess a version check is needed?

I guess this will break older versions.

File: changelogs/fragments/7880-ipa-fix-sudo-and-hbcalrule-idempotence.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+bugfixes:
+  - ipa_sudorule - the module uses a string for ``ipaenabledflag`` while the returned value is a boolean
+  - ipa_hbacrule - the module uses a string for ``ipaenabledflag`` while the returned value is a boolean
```
Comment (felixfontein) [inline changelogs/fragments/7880-ipa-fix-sudo-and-hbcalrule-idempotence.yml:3]:
```suggestion
  - ipa_sudorule - the module uses a string for ``ipaenabledflag`` for new FreeIPA versions while the returned value is a boolean (https://github.com/ansible-collections/community.general/pull/7880).
  - ipa_hbacrule - the module uses a string for ``ipaenabledflag`` for new FreeIPA versions while the returned value is a boolean (https://github.com/ansible-collections/community.general/pull/7880).
```

**Review by felixfontein — COMMENTED** [review]
If nobody objects, I'll merge this in ~a week.

**Comment by felixfontein** [comment]
@aBUDmdBQ thanks for your contribution!

**Pull request merged**